contracts/random/random.go

```diff
@@ -0,0 +1,203 @@
+			"type": "function"
+		},
+		{
+			"constant": true,
+			"inputs": [],
+			"name": "random",
+			"outputs": [
+				{
+					"name": "",
+					"type": "bytes32"
+				}
+			],
+			"payable": false,
+			"stateMutability": "view",
+			"type": "function"
+		},
+		{
+			"constant": true,
+			"inputs": [
+				{
+					"name": "blockNumber",
+					"type": "uint256"
+				}
+			],
+			"name": "getBlockRandomness",
+			"outputs": [
+				{
+					"name": "",
+					"type": "bytes32"
+				}
+			],
+			"payable": false,
+			"stateMutability": "view",
+			"type": "function"
+		}
+	]`
+)
+
+var (
+	revealAndCommitMethod    *contracts.BoundMethod
+	commitmentsMethod        *contracts.BoundMethod
+	computeCommitmentMethod  *contracts.BoundMethod
+	randomMethod             *contracts.BoundMethod
+	getBlockRandomnessMethod *contracts.BoundMethod
+)
+
+func init() {
+	randomAbi, err := abi.JSON(strings.NewReader(randomABI))
+	if err != nil {
+		panic(err)
+	}
+
+	revealAndCommitMethod = contracts.NewRegisteredContractMethod(params.RandomRegistryId, &randomAbi, "revealAndCommit", params.MaxGasForRevealAndCommit)
+	commitmentsMethod = contracts.NewRegisteredContractMethod(params.RandomRegistryId, &randomAbi, "commitments", params.MaxGasForCommitments)
+	computeCommitmentMethod = contracts.NewRegisteredContractMethod(params.RandomRegistryId, &randomAbi, "computeCommitment", params.MaxGasForComputeCommitment)
+	randomMethod = contracts.NewRegisteredContractMethod(params.RandomRegistryId, &randomAbi, "random", params.MaxGasForBlockRandomness)
+	getBlockRandomnessMethod = contracts.NewRegisteredContractMethod(params.RandomRegistryId, &randomAbi, "getBlockRandomness", params.MaxGasForBlockRandomness)
+}
+
+func IsRunning(vmRunner vm.EVMRunner) bool {
+	randomAddress, err := contracts.GetRegisteredAddress(vmRunner, params.RandomRegistryId)
+
+	if err == contracts.ErrSmartContractNotDeployed || err == contracts.ErrRegistryContractNotDeployed {
+		log.Debug("Registry address lookup failed", "err", err, "contract", hexutil.Encode(params.RandomRegistryId[:]))
+	} else if err != nil {
+		log.Error(err.Error())
+	}
+
+	return err != nil && randomAddress != common.ZeroAddress
+}
+
+// GetLastCommitment returns up the last commitment in the smart contract
+func GetLastCommitment(vmRunner vm.EVMRunner, validator common.Address) (common.Hash, error) {
+	lastCommitment := common.Hash{}
+	err := commitmentsMethod.Query(vmRunner, &lastCommitment, validator)
+	if err != nil {
+		log.Error("Failed to get last commitment", "err", err)
+		return lastCommitment, err
+	}
+
+	if (lastCommitment == common.Hash{}) {
+		log.Debug("Unable to find last randomness commitment in smart contract")
+	}
+
+	return lastCommitment, nil
+}
+
+// ComputeCommitment calulcates the commitment for a given randomness.
+func ComputeCommitment(vmRunner vm.EVMRunner, randomness common.Hash) (common.Hash, error) {
+	commitment := common.Hash{}
+	// TODO(asa): Make an issue to not have to do this via StaticCall
```

Comment:
I don't know what this meant.  Ideally we'd see what this was about and take care of it and/or remove the TODO.  If not, at least update the TODO to not reference `StaticCall` since now it uses vmRunner

Reply:
Asking @asaj to see if he remember or if we can remove it